base/abstractarraymath.jl

```diff
@@ -34,6 +34,7 @@ julia> vec(a)
 """
 vec(a::AbstractArray) = reshape(a,_length(a))
 vec(a::AbstractVector) = a
+vec(a1::AbstractArray, a2::AbstractArray, arest::AbstractArray...) = vcat(vec(a1), vec(a2), map(vec, arest)...)
```

Comment:
`vec` returns a view for `Array`, and a `ReshapedArray` by default for other `AbstractArray` types.  But in principle it isn't a view for all types.